python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -172,11 +233,10 @@ def compare_alleles(row: pd.Series, col1: str, col2: str) -> GwasStatsLocusKind:
     return GwasStatsLocusKind.NO_MATCH
 
 
-def adjust_dosages(row: pd.Series) -> pd.Series:
-    """Adjust the dosage for if the effect allele is not the alternate allele."""
-    if row["allele_comparison"] == "effect allele is ref":
+def adjust_dosages(row):
+    if row["allele_comparison"] == "reversed match":
         for col in row.index:
-            if col != "allele_comparison":
+            if col != "allele_comparison" and row[col] != -1:
```

Comment:
I would leave this a bit more general to work with null missing values as well: https://github.com/bystrogenomics/bystro/blob/master/python/python/bystro/ancestry/inference.py#L276